tajo-plan/src/main/java/org/apache/tajo/plan/rewrite/SelfDescSchemaBuildPhase.java

```diff
@@ -0,0 +1,479 @@
+        node.setInSchema(child.getOutSchema());
+        node.setOutSchema(node.getInSchema());
+      }
+      return node;
+    }
+
+    @Override
+    public LogicalNode visitInsert(ProcessorContext ctx, Stack<Expr> stack, Insert expr) throws TajoException {
+      super.visitInsert(ctx, stack, expr);
+
+      InsertNode node = getNodeFromExpr(ctx.planContext.getPlan(), expr);
+      LogicalNode child = getNonRelationListExpr(ctx.planContext.getPlan(), expr.getSubQuery());
+      node.setInSchema(child.getOutSchema());
+      node.setOutSchema(node.getInSchema());
+      return node;
+    }
+
+    @Override
+    public LogicalNode visitRelation(ProcessorContext ctx, Stack<Expr> stack, Relation expr) throws TajoException {
+      LogicalPlan plan = ctx.planContext.getPlan();
+      QueryBlock queryBlock = plan.getBlockByExpr(expr);
+      ScanNode scan = queryBlock.getNodeFromExpr(expr);
+      TableDesc desc = scan.getTableDesc();
+
+      if (desc.hasSelfDescSchema()) {
+        if (ctx.projectColumns.containsKey(getQualifiedRelationName(ctx.planContext, expr))) {
+          Set<Column> columns = new HashSet<>();
+          for (ColumnReferenceExpr col : ctx.projectColumns.get(getQualifiedRelationName(ctx.planContext, expr))) {
+            columns.add(NameResolver.resolve(plan, queryBlock, col, NameResolvingMode.RELS_ONLY, true));
+          }
+
+          desc.setSchema(buildSchemaFromColumnSet(columns));
+          scan.init(desc);
+        } else {
+          // error
+          throw new TajoInternalError(
+              "Columns projected from " + getQualifiedRelationName(ctx.planContext, expr) + " is not found.");
+        }
+      }
+
+      return scan;
+    }
+
+    /**
+     * This method creates a schema from a set of columns.
+     * For a nested column, its ancestors are guessed and added to the schema.
+     * For example, given a column 'glossary.title', the columns of (glossary RECORD (title TEXT)) will be added
+     * to the schema.
+     *
+     * @param columns a set of columns
+     * @return schema build from columns
+     */
+    private Schema buildSchemaFromColumnSet(Set<Column> columns) {
+      SchemaGraph schemaGraph = new SchemaGraph();
+      Set<ColumnVertex> rootVertexes = new HashSet<>();
+      Schema schema = new Schema();
+
+      for (Column eachColumn : columns) {
+        String simpleName = eachColumn.getSimpleName();
+        if (NestedPathUtil.isPath(simpleName)) {
+          String[] paths = simpleName.split(NestedPathUtil.PATH_DELIMITER);
+          for (int i = 0; i < paths.length-1; i++) {
+            String parentName = paths[i];
+            if (i == 0) {
+              parentName = CatalogUtil.buildFQName(eachColumn.getQualifier(), parentName);
+            }
+            // Leaf column type is TEXT; otherwise, RECORD.
+            Type childDataType = (i == paths.length-2) ? Type.TEXT : Type.RECORD;
```

Comment:
These lines exceed 120 column width. In addition, ``visitCreateTable()`` in this file exceeds 120 too.